control-plane-operator/controllers/hostedcontrolplane/ignition/reconcile.go

```diff
@@ -322,3 +336,51 @@ func fileFromBytes(path string, mode int, contents []byte) igntypes.File {
 		},
 	}
 }
+
+func generateKubernetesDefaultProxyPod(image string, listenAddr string, proxyAddr string, apiserverAddr string) func() ([]byte, error) {
+	return func() ([]byte, error) {
+		p := &corev1.Pod{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: corev1.SchemeGroupVersion.String(),
+				Kind:       "Pod",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "kube-apiserver-proxy",
+				Namespace: "kube-system",
+				Labels: map[string]string{
+					"k8s-app": "kube-apiserver-proxy",
+					"hypershift.openshift.io/control-plane-component": "kube-apiserver-proxy",
+				},
+			},
+			Spec: corev1.PodSpec{
+				HostNetwork:       true,
+				PriorityClassName: "system-node-critical",
+				Containers: []corev1.Container{{
+					Image: image,
+					Name:  "kubernetes-default-proxy",
+					Command: []string{
+						"control-plane-operator",
+						"kubernetes-default-proxy",
+						"--listen-addr=" + listenAddr,
+						"--proxy-addr=" + strings.TrimPrefix(strings.TrimPrefix(proxyAddr, "http://"), "https://"),
+						"--apiserver-addr=" + apiserverAddr,
+					},
+					SecurityContext: &corev1.SecurityContext{
+						RunAsUser: pointer.Int64Ptr(config.DefaultSecurityContextUser),
+					},
+					Resources: corev1.ResourceRequirements{
+						Requests: corev1.ResourceList{
+							corev1.ResourceMemory: resource.MustParse("16Mi"),
```

Comment:
should we use more typical memory/cpu here? Seems that a go binary might use more resources than haproxy

Reply:
Because the networking stack doesn't work yet because it doesn't plug the proxy config through the prometheus doesn't run so something like `kubectl top` doesn't work so I can't really tell.

Can we update this later on as needed once things generally work?

Reply:
sure